closed/OpenJ9.gmk

```diff
@@ -28,11 +28,22 @@ ifeq (,$(BUILD_ID))
   BUILD_ID := 000000
 endif
 
+OPENJ9_BRANCH := $(shell git -C $(OPENJ9_TOPDIR)    rev-parse --abbrev-ref HEAD)
+# find the closest tag name
+OPENJ9_TAG    := $(shell git -C $(OPENJ9_TOPDIR)    describe --abbrev=0)
 OPENJ9_SHA    := $(shell git -C $(OPENJ9_TOPDIR)    rev-parse --short HEAD)
 OPENJ9OMR_SHA := $(shell git -C $(OPENJ9OMR_TOPDIR) rev-parse --short HEAD)
 ifeq (,$(OPENJ9_SHA))
   $(error Could not determine OpenJ9 SHA)
 endif
+ifeq (,$(OPENJ9_TAG))
+ifeq (,$(OPENJ9_BRANCH))
+  $(error Could not determine OpenJ9 Branch)
+endif
+OPENJ9_VERSION_STRING := $(OPENJ9_BRANCH)-$(OPENJ9_SHA)
+else
+OPENJ9_VERSION_STRING := $(OPENJ9_TAG)
+endif
```

Comment:
I'd like to see this reorganized so it doesn't compute OPENJ9_BRANCH needlessly:

```
ifneq (,$(OPENJ9_TAG))
  OPENJ9_VERSION_STRING := $(OPENJ9_TAG)
else
  OPENJ9_BRANCH := $(shell git -C $(OPENJ9_TOPDIR) rev-parse --abbrev-ref HEAD)
  ifneq (,$(OPENJ9_BRANCH))
    OPENJ9_VERSION_STRING := $(OPENJ9_BRANCH)-$(OPENJ9_SHA)
  else
    $(error Could not determine OpenJ9 Branch)
  endif
endif
```

Reply:
Fixed.